arangod/VocBase/LogicalCollection.cpp

```diff
@@ -1298,7 +1298,28 @@ std::optional<replication2::LogId> LogicalCollection::tryShardIdToStateId(
   return replication2::LogId::fromString(stateId);
 }
 
-auto LogicalCollection::getDocumentState()
+auto LogicalCollection::isLeadingShard() const -> bool {
+  if (!ServerState::instance()->isDBServer()) {
+    // Only DBServers can be leaders of shards
+    return false;
+  }
+  if (isAStub()) {
+    // Stubs are no shards, they cannot be leaders
+    return false;
+  }
+  if (replicationVersion() == replication::Version::ONE) {
+    // Replication version 1 does not have leaders
+    auto const& followerInfo = followers();
+    // If the Leader is empty, we are the leader
+    return followerInfo->getLeader().empty();
+  } else {
+    auto const& docState = getDocumentState();
```

Comment:
This method works. I want to point out that `getDocumentState()` may throw if the replicated log is not there. I doubt this is possible with the current use-case, as the transaction is already there.

A way to make sure we would never crash is to wrap this into a `catchToResultT`. You may use `LogicalCollection::getDocumentStateLeader()` for that, which should always result in `ok` if the leader is there.